src/prommis/uky/costing/models/hd_ce.py

```diff
@@ -0,0 +1,1053 @@
+
+
+def custom_REE_plant_currency_units():
+    """
+    Define conversion rates for US Dollars based on CE Index.
+    """
+    register_idaes_currency_units()
+    if "USD_Jan_2024" in units._pint_registry:  # pylint: disable=protected-access
+        # Assume that custom REE plant units have already been registered
+        # Log a message and end
+        _log.debug(
+            "Custom REE plant currency units (USD_Jan_2024) "
+            "already appear in Pyomo unit registry. Assuming repeated "
+            "call of custom_power_plant_currency_units."
+        )
+    else:
+        units.load_definitions_from_strings(
+            [
+                # from https://toweringskills.com/financial-analysis/cost-indices/ as of 01/02/2024
+                "USD_Jan_2024 = 500/795.4 * USD_CE500",
+            ]
+        )
+
+
+@declare_process_block_class("REEEquipmentCosting")
+class REEEquipmentCostingData(FlowsheetCostingBlockData):
+    # Register currency and conversion rates based on CE Index
+    custom_REE_plant_currency_units()
+
+    def build_global_params(self):
+        # Set the base year for all costs
+        self.base_currency = units.USD_Jan_2024
+        # Set a base period for all operating costs
+        self.base_period = units.year
+
+    def cost_hydrogen_decrepitation_furnace(
+        blk,
+        ramp_up_time=300,  # (in seconds)
+        operating_temperature=443.15,  # in Kelvin
+        decrepitation_duration=10800,  # (in seconds)
+        preparation_time=3600,  # in seconds
+        cool_down_time=3600,  # in seconds
+        sample_heat_capacity=0.44,  # in kJ/(kg*K)
+        sample_mass=None,  # in kg
+        sample_volume=None,  # in m**3
+        sample_density=7500,  # in kg/m**3
+        chamber_to_sample_ratio=2,
+        length_insulation1=7.62,  # m
+        width_insulation1=0.6096,  # m
+        thickness_insulation1=0.0254,  # m
+        price_insulation1=183.81,  # USD
+        weight_insulation1=15.42,  # kg
+        thermal_cond_insulation_material1=0.33,  # W/(m*K)
+        price_metal1=3.14,  # USD/kg
+        density_metal1=7473.57,  # kg/m**3
+        thermal_cond_metal_material1=13.53,  # W/(m*K)
+        length_insulation2=1.19,  # m
+        width_insulation2=0.381,  # m
+        thickness_insulation2=0.0889,  # m
+        price_insulation2=47.00,  # USD
+        weight_insulation2=10.43,  # kg
+        thermal_cond_insulation_material2=0.069,  # W/(m*K)
+        price_metal2=1.50,  # USD/kg
+        density_metal2=7861.09,  # kg/m**3
+        thermal_cond_metal_material2=45,  # W/(m*K)
+        hours_per_shift=8,
+        shifts_per_day=3,
+        specific_heat_capacity_insulation1=1.08,  # KJ/(kg*K)
+        specific_heat_capacity_metal1=0.468,  # KJ/(kg*K)
+        specific_heat_capacity_insulation2=0.9,  # KJ/(kg*K)
+        specific_heat_capacity_metal2=0.502416,  # KJ/(kg*K)
+        operating_days_per_year=336,
+        efficiency=0.95,
+        electricity_rate=0.081,  # USD/Kwhr
```

Comment:
A note here - it might be better to explicitly include units of measurement here for verification and conversion purposes rather than have an implicit assumption. If you do it right, this means that the value users provide will be automatically converted to the desired units, no matter what units they use.

I.e. `electricity_rate=0.081*units.USDXXXX/units.kW/units.hour`